### PR TITLE
Fix ChunkWatchEvent.(Un)Watch not firing correctly

### PR DESCRIPTION
**Pull request by malte0811**
As described in #6245, the events are currently only firing for the chunks (un)watched on joining/leaving the world.

I'll make a 1.14 version of this once the 1.15 version is merged since I don't want to have to apply every fix twice, and the relevant code doesn't seem to have changed.

### CODE REVIEW

**Comment by ichttt** [comment]
Can you please resolve the merge conflicts?

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 1]
```diff
@@ -686,11 +686,11 @@ public static boolean saplingGrowTree(IWorld world, Random rand, BlockPos pos)
         return event.getResult() != Result.DENY;
     }
 
-    public static void fireChunkWatch(boolean watch, ServerPlayerEntity entity, ChunkPos chunkpos, ServerWorld world)
+    public static void fireChunkWatch(boolean wasLoaded, boolean load, ServerPlayerEntity entity, ChunkPos chunkpos, ServerWorld world)
```
Comment (pupnewfster) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: This is more a question for the forge team than you directly, but does this need some sort of fallback given technically this is a "breaking change", and now that forge has an RB.

Comment (gigaherz) [reply src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: Normally the answer would be yes, however this is an event factory used within the chunk manager. No mod should be calling this. Lex has the last word as always, but this doesn't feel like it conflicts with the "no breaking changes" rule.

Comment (LexManos) [reply src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: Yes it falls under binary compatibility

**Pull request merged**